force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -36,16 +36,22 @@ public with sharing class DummyJSONCallout {
 
 	public static void getDummyJSONUserFromId(String dummyUserId) {
 		// Create HTTP request to send.
-
+		HttpRequest request = new HttpRequest();
 		// Set the endpoint URL. Use direct URL or for best practices use Named Credential.
-
+		request.setEndpoint('https://dummyjson.com/user/' + dummyUserId);
 		// Set the HTTP method to GET.
-
+		request.setMethod('GET');
 		// Send the HTTP request and get the response.
-
+		Http http = new Http();
+		HttpResponse response = http.send(request);
 		// If the HTTP response code is successful, parse the JSON response and update the contact.
 		Contact cont = null;
-		upsert cont DummyJSON_Id__c; //insert/update from the JSON response using the external id (dummyUserId)
+		if (response.getStatusCode() >= 200 || response.getStatusCode() <= 299) {
+			cont = createContactFromJson(response.getBody());
+			cont.DummyJSON_Id__c = dummyUserId;
+		}
+		Schema.SObjectField ftoken = Contact.Fields.DummyJSON_Id__c;
+		Database.upsert(cont, ftoken, AccessLevel.USER_MODE); //insert/update from the JSON response using the external id (dummyUserId)
```

Comment:
Is there a simpler way to do this without a Database class?
Try doing a direct upsert